kubernetes/src/main/java/io/kubernetes/client/common/PartialObjectMetadata.java

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.common;
+
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+public class PartialObjectMetadata implements KubernetesObject {
```

Comment:
Should we templatize this for clarity? so that someone can declare a:

`PartialObjectMetadata<V1Pod>` or somesuch, so that you can really easily convert it into it's actual object?

Reply:
Yeah, but I was imagining a world where you got the metadata but wanted to pass it off to something that knew how to print Pods or somesuch, it's definitely not required, but I kind of like the idea of knowing what kind of thing I have gotten the Metadata for from a type-safety perspective...

Reply:
Done... I've removed POM and POML classes. You are right that you can just use the original, specific classes.